src/components/common/Map/Map.js

```diff
@@ -16,29 +29,86 @@ const Map = ({
 }) => {
   const [locationStatus, setLocationStatus] = useState('');
   const [currentLocation, setCurrentLocation] = useState(defaultCenter);
+  const getAllTargetsRequest = useDispatch(getAllTargets);
+  const { targets } = useTargets();
 
   useEffect(() => {
     const success = ({ coords: { latitude, longitude } }) => {
       setCurrentLocation({ lat: latitude, lng: longitude });
     };
     const error = () => setLocationStatus(<FormattedMessage id="home.current_location_failed" />);
     navigator.geolocation.getCurrentPosition(success, error);
-  }, [locationStatus]);
+    getAllTargetsRequest();
+  }, [locationStatus, getAllTargetsRequest]);
+
+  const targetIcon = target => {
+    switch (target.topicId) {
+      case 2:
+        return FootballTopicIcon;
+      case 13:
+        return PoliticsTopicIcon;
+      case 14:
+        return ArtTopicIcon;
+      case 15:
+        return DatingTopicIcon;
+      case 16:
+        return MusicTopicIcon;
+      case 17:
+        return MoviesTopicIcon;
+      case 18:
+        return SeriesTopicIcon;
+      case 19:
+        return FoodTopicIcon;
+      default:
+        return MusicTopicIcon;
+    }
+  };
+
+  const handleTargetsCircles = ({ map, maps }) => {
+    return targets.map(({ target }) => {
```

Comment:
```suggestion
    return targets.map(({ target: { lat, lng, radius, topicId } }) => {
```
I think you can destructure here and then use the values directly